graylog2-web-interface/src/stores/metrics/MetricsStore.js

```diff
@@ -169,6 +170,11 @@ const MetricsStore = Reflux.createStore({
 
       return fetch('GET', url).then((response) => {
         return { nodeId: nodeId, names: response.metrics };
+      }, (error) => {
+        UserNotification.error(`Fetching metrics for node [${nodeId}] failed with: ${error}.`,
+          'Could not fetch node metrics');
+        // When fetching metrics fails, keep previous available metrics around, letting user see them
+        return { nodeId: nodeId, names: this.metricsNames[nodeId], error: error };
```

Comment:
We are returning `error` from the promise handler there. Do we want to make use of that in the error handling part of the `ShowMetricsPage` and show the error there?

Reply:
I guess we could replace the toastr error notification with something less annoying, will take a quick look 👍

Reply:
@dennisoelkers I reworked the UI a bit to display a text error next to the filter, instead of possibly multiple toaster notifications. I'll re-request for your review, let me know what you think :)

Reply:
When I am going to the node metrics page with your patch applied, I end up [here]( https://github.com/Graylog2/graylog2-server/pull/9316/files#diff-7a1cae0ff3ded5ec2c3534169b8fd9650594e6a0cd1f848b272d69e4069383b8R66) (`ShowMetricsPage.jsx:66`), which does not show the error message of the error that occurred. Maybe we should centralize error handling in one of the two components (`ShowMetricsPage` vs. `MetricsComponent`) instead of doing it in both?

Reply:
I think you refer to the change in the page header for `ShowMetricsPage` (the link takes me somewhere else, somehow). I was thinking about putting the error message in the page header, but decided against it since it may take quite a bit of space and the error message will be displayed below anyway.

Right now all of the error handling is done in `MetricsComponent`, since it's the component rendering the page content and seemed cleaner that way (less imports and need to move stuff around). The only exception is the page header in `ShowMetricsPage`, and only because we indicate the number of metrics the node is reporting.